app/code/Magento/Inventory/Test/Integration/Indexer/StockIndexerTest.php

```diff
@@ -99,4 +99,18 @@ public function testReindexAll()
         self::assertEquals(5, $this->getProductQuantityInStock->execute('SKU-2', 20));
         self::assertEquals(5, $this->getProductQuantityInStock->execute('SKU-2', 30));
     }
+
+    /**
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/sources.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stocks.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source_items.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stock_source_link.php
+     */
+    public function testReindexZeroQuantityIfSourceItemIsOutOfStock()
```

Comment:
Looks like need to move this case in SourceItemIndexerTest  (or in separate class)
Because described case is related to source items indexing logic but not to stock indexing